### PR TITLE
Make memory-safety instrumentations copy dbg metadata

### PR DESCRIPTION
**Pull request by shaobo-he**

### CODE REVIEW

**Comment by zvonimir** [comment]
What does this do? I don't quite get it.

**Comment by shaobo-he** [comment]
> What does this do? I don't quite get it.

This is to provide more precise error traces for memory-safety violations. If you run the regressions in this PR using the develop branch, you should notice that the error trace is weird.

**Comment by zvonimir** [comment]
What happened with CI here?

File: test/c/memory-safety/test_sourceloc.c [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#include "smack.h"
+#include <stdlib.h>
+
+// @expect verified
+
+int main() {
```
Comment (zvonimir) [inline test/c/memory-safety/test_sourceloc.c:6]: This should be `int main(void)`

File: test/c/memory-safety/test_sourceloc_fail.c [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#include "smack.h"
+#include <stdlib.h>
+
+// @expect error
+// @checkout grep -v "test_sourceloc_fail.c(16,9)"
+
+int main() {
```
Comment (zvonimir) [inline test/c/memory-safety/test_sourceloc_fail.c:7]: This should be `int main(void)`

**Review by zvonimir — CHANGES_REQUESTED** [review]

**Comment by shaobo-he** [comment]
> What happened with CI here?

It's related to this issue: https://github.com/smackers/smack/issues/668

**Comment by zvonimir** [comment]
Hey @shaobo-he . I can't merge this until CI is fixed. What is happening there?

**Comment by zvonimir** [comment]
How is it related to the issue you pointed out? I don't understand how come that regression works now?

**Comment by zvonimir** [comment]
Ah, I see. You added a regression! Ok, issue #668 might be an easy fix. Can you take care of that one first then? Or how should we go about this?

**Comment by shaobo-he** [comment]
> Ah, I see. You added a regression! Ok, issue #668 might be an easy fix. Can you take care of that one first then? Or how should we go about this?

Yes, I think I'll fix it first.

**Comment by shaobo-he** [comment]
@zvonimir please see https://github.com/smackers/smack/pull/670.

**Review by zvonimir — APPROVED** [review]
Looks good to me.

**Pull request merged**